src/libstd/io/mod.rs

```diff
@@ -354,19 +354,26 @@ fn append_to_string<F>(buf: &mut String, f: F) -> Result<usize>
 // avoid paying to allocate and zero a huge chunk of memory if the reader only
 // has 4 bytes while still making large reads if the reader does have a ton
 // of data to return. Simply tacking on an extra DEFAULT_BUF_SIZE space every
-// time is 4,500 times (!) slower than this if the reader has a very small
```

Comment:
This doesn't look like a good justification, generally optimisation should focus big size, small are 4500 slower but 0.1ns * 4500 isn't much.

That said, the caller could reserve a big vector itself, but that only work for the first time. Is Rust allocation behaviour to increase a lot the size even if only a small amount is requested ? like 32 => 64 => 128 => 256 => etc even if only 32 is ask each time ?

Reply:
As with the number `32`, this was not my design choice; I was only fixing https://github.com/rust-lang/rust/issues/51746 - all the rest was left as-is.